src/main/java/org/opensearch/ad/indices/AnomalyDetectionIndices.java

```diff
@@ -281,10 +294,132 @@ public boolean doesAnomalyDetectorJobIndexExist() {
+                        logger.error(error);
+                        listener.onFailure(new EndRunException(error, true));
+                    }
+                }, exception -> {
+                    if (ExceptionsHelper.unwrapCause(exception) instanceof ResourceAlreadyExistsException) {
+                        // It is possible the index has been created while we sending the create request
+                        validateCustomResultIndexAndExecute(resultIndex, function, listener);
+                    } else {
+                        logger.error("Failed to create anomaly detector result index " + resultIndex, exception);
+                        listener.onFailure(exception);
+                    }
+                }));
+            } else {
+                validateCustomResultIndexAndExecute(resultIndex, function, listener);
+            }
+        } catch (Exception e) {
+            logger.error("Failed to create detector result index " + resultIndex, e);
+            listener.onFailure(e);
+        }
+    }
+
+    private <T> void validateCustomResultIndexAndExecute(String resultIndex, AnomalyDetectorFunction function, ActionListener<T> listener) {
+        try {
+            if (!isValidResultIndex(resultIndex)) {
+                logger.warn("Can't create detector with custom result index {} as its mapping is invalid", resultIndex);
+                listener.onFailure(new IllegalArgumentException("Invalid result index: " + resultIndex));
+                return;
+            }
+
+            AnomalyResult dummyResult = AnomalyResult.getDummyResult();
+            IndexRequest indexRequest = new IndexRequest(resultIndex)
+                .id(DUMMY_AD_RESULT_ID)
+                .source(dummyResult.toXContent(XContentBuilder.builder(XContentType.JSON.xContent()), ToXContent.EMPTY_PARAMS));
+            client.index(indexRequest, ActionListener.wrap(response -> {
+                logger.debug("Successfully wrote dummy AD result to result index {}", resultIndex);
+                client.delete(new DeleteRequest(resultIndex).id(DUMMY_AD_RESULT_ID), ActionListener.wrap(deleteResponse -> {
+                    logger.debug("Successfully deleted dummy AD result from result index {}", resultIndex);
+                    function.execute();
+                }, ex -> {
+                    logger.error("Failed to delete dummy AD result from result index " + resultIndex, ex);
+                    listener.onFailure(ex);
+                }));
+            }, exception -> {
+                logger.error("Failed to write dummy AD result to result index " + resultIndex, exception);
+                listener.onFailure(exception);
+            }));
+        } catch (Exception e) {
+            logger.error("Failed to create detector with custom result index " + resultIndex, e);
+            listener.onFailure(e);
+        }
+    }
```

Comment:
Seems weird to be indexing and deleting to the custom index. Why isn't the check on the index mappings in `isValidResultIndex` enough here?

Reply:
User may have no write permission on custom result index. Talked with security plugin team, seems no easy way to verify if user has write permission. So just tried to write and delete a dummy anomaly result to verify.

Reply:
Yeah, just seems weird, in case it fails to delete for example, and the user sees some dummy result mixed in with their results. Have you considered security owning validation on write permissions? Seems more appropriate to be owned by them.

Obviously with the given timeline, this is fine with me (or even removing the write check altogether). Just thinking for future use cases.

Reply:
Right, by "security owning validation on write permissions", I meant more about improved functionality on the security end to have such an interface to check index read/write permissions. In that case, it could just be an extra check during detector creation, where the result index permissions are checked, to see if such user has write permissions.

Since that's not available, I'm ok with the current way of injecting and removing dummy values. Agreed it is probably best to fail during creation phase than write anomaly result phase, where it would be more work for the user.